File: src/main/java/com/tram/springbootangularboard/security/token/PostAuthorizationToken.java
package com.tram.springbootangularboard.security.token;

import com.tram.springbootangularboard.security.AccountContext;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

public class PostAuthorizationToken extends UsernamePasswordAuthenticationToken {

    public PostAuthorizationToken(Object principal, Object credentials, Collection<? extends GrantedAuthority> authorities) {
        super(principal, credentials, authorities);
    }

    public AccountContext getAccountContext() {
        return (AccountContext)super.getPrincipal();
    }

    public static PostAuthorizationToken getFromAccountContext(AccountContext accountContext) {
        return new PostAuthorizationToken(accountContext, accountContext.getPassword(), accountContext.getAuthorities());
    }
}

File: src/main/java/com/tram/springbootangularboard/security/tokens/PostAuthorizationToken.kt
package com.tram.springbootangularboard.security.tokens

import com.tram.springbootangularboard.domain.Account
import com.tram.springbootangularboard.security.AccountContext
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority

class PostAuthorizationToken : UsernamePasswordAuthenticationToken{
    constructor(principal: Any?, credentials: Any?, authorities: MutableCollection<out GrantedAuthority>?) : super(principal, credentials, authorities)
    companion object {
        fun getTokenFromAccountContext(accountContext: AccountContext): PostAuthorizationToken {
            return PostAuthorizationToken(accountContext, accountContext.password, accountContext.authorities)
        }
    }
}
